assets/kube-state-metrics/deployment.yaml

```diff
@@ -108,6 +124,7 @@ spec:
         - --tls-private-key-file=/etc/tls/private/tls.key
         - --client-ca-file=/etc/tls/client/client-ca.crt
         - --config-file=/etc/kube-rbac-policy/config.yaml
+        - --ignore-paths=/metrics
```

Comment:
not sure to understand why we ignore /metrics here?

Reply:
Right, even though we've always used telemetry metrics to check for readiness upstream, nonetheless, exposing that information is susceptible to malicious intent. I'll patch upstream to restrict that information from being exposed and add a dedicated `/readyz`.

Reply:
On second thought, it'd be better to not rely on telemetry endpoints at all and base all probes off of the actual KSM resource metrics server.

Reply:
Opened https://github.com/kubernetes/kube-state-metrics/pull/2442.